src/github.com/couchbase/sync_gateway/db/database.go

```diff
@@ -197,6 +201,8 @@ func (dc *DatabaseContext) TakeDbOffline() error {
 
 func (context *DatabaseContext) Authenticator() *auth.Authenticator {
 	// Authenticators are lightweight & stateless, so it's OK to return a new one every time
+	context.BucketLock.RLock()
+	defer context.BucketLock.RUnlock()
```

Comment:
This comment from [golang docs](https://golang.org/doc/articles/race_detector.html) is relevant, we are only detecting race conditions from code paths tested via out unit tests.

```
The race detector only finds races that happen at runtime, so it can't find races in code paths that are not executed. If your tests have incomplete coverage, you may find more races by running a binary built with -race under a realistic workload.
```

Current unit test coverage using test -cover

```
ok  	github.com/couchbase/sync_gateway/auth	24.854s	coverage: 60.2% of statements
ok  	github.com/couchbase/sync_gateway/base	1.042s	coverage: 10.8% of statements
ok  	github.com/couchbase/sync_gateway/channels	1.714s	coverage: 66.7% of statements
ok  	github.com/couchbase/sync_gateway/db	18.121s	coverage: 71.7% of statements
ok  	github.com/couchbase/sync_gateway/rest	84.806s	coverage: 58.5% of statements
```